packages/components/src/components/SquiggleViewer/VariableBox.tsx

```diff
@@ -62,12 +74,31 @@ export const VariableBox: FC<VariableBoxProps> = ({
   const [, forceUpdate] = useReducer((x) => x + 1, 0);
 
   const { path } = value;
+  const isRoot = Boolean(path?.isRoot());
+  const childrenElements = getChildrenValues(value);
+  const initialSettings = makeInitialSettings(childrenElements.length, isRoot);
 
   if (!path) {
     throw new Error("Can't display a pathless value");
   }
 
-  const settings = getSettings(path);
+  initialSettings.collapseChildren && collapseChildren(value);
```

Comment:
I'm surprised that this works as well as does. This part is complicated, kudos for making it work.

There are a few corner cases where if you define one variable and then replace it with another variable (e.g., `List.upTo(1,20)` -> `List.upTo(1,3)`), you get some remnants from the old version (children will stay collapsed). But that's probably unavoidable and it's not important.

Imperative code in the component's body worries me a bit, but since it modifies only `ref` values, I guess it's fine.

Reply:
Yea, I was also nervous about it. But I couldn't think of other ways of doing it that wouldn't be a whole lot of work.